src/SonarQube.Client/Services/RequestFactory.cs

```diff
@@ -98,7 +98,7 @@ public TRequest Create<TRequest>(Version version)
             SortedList<Version, Func<IRequest>> map;
             if (registrations.TryGetValue(typeof(TRequest), out map))
             {
-                Log($"Looking up implementation of '{typeof(TRequest).Name}' for version '{version}'.");
+                Log($"Looking up implementation of '{typeof(TRequest).Name}' for version '{version}' on thread '{System.Threading.Thread.CurrentThread.ManagedThreadId}'");
```

Comment:
Miscellaneous: extra debug information. It helped show a race condition that can occur while connecting (#696)